locations/spiders/bim_tr.py

```diff
@@ -0,0 +1,46 @@
+from scrapy import FormRequest, Spider
+
+from locations.categories import Categories, apply_category
+from locations.items import Feature
+from locations.pipelines.address_clean_up import clean_address
+
+
+class BimTRSpider(Spider):
+    name = "bim_tr"
+    item_attributes = {"brand": "BİM", "brand_wikidata": "Q1022075"}
+    start_urls = ["https://www.bim.com.tr/Categories/104/magazalar.aspx"]
+    no_refs = True
+
+    def parse(self, response, **kwargs):
+        city_list = response.xpath('//select[@id="BimFiltre_DrpCity"]/option')
+        for city in city_list:
+            city_name = city.xpath("./text()").get()
+            city_key = city.xpath("./@value").get()
+            yield FormRequest(
+                url=response.url,
+                formdata={"BimFiltre$DrpCity": city_key},
+                callback=self.parse_county,
+                cb_kwargs=dict(city_name=city_name, city_key=city_key),
+            )
+
+    def parse_county(self, response, **kwargs):
+        county_list = response.xpath('//select[@id="BimFiltre_DrpCounty"]/option')
+        for county in county_list:
+            county_key = county.xpath("./@value").get()
+            yield FormRequest(
+                url=response.url,
+                method="GET",
+                formdata={"CityKey": kwargs["city_key"], "CountyKey": county_key},
+                callback=self.parse_stores,
+                cb_kwargs=kwargs,
+            )
+
+    def parse_stores(self, response, **kwargs):
+        for store in response.xpath('//*[@class="detailArea"]//*[@class="title"]'):
+            item = Feature()
+            item["branch"] = store.xpath("./text()").get()
+            item["addr_full"] = clean_address(store.xpath("./following::p/text()").get())
```

Comment:
Addresses are a bit inconsistent, in most cases it's full address, but in some big cities it's street address ([examples](https://www.bim.com.tr/Categories/104/magazalar.aspx?CityKey=7&CountyKey=2047&firin=0)). 

May I kindly ask to propagate county text value from `parse_county` to `parse_stores` then adjust addresses a bit? In the end we should end up with better full addresses. Logic could be like:

```python
item["addr_full"] = clean_address(store.xpath("./following::p/text()").get())

if kwargs["county"] not in item["addr_full"]:
    item["addr_full"] = item["addr_full"] + kwargs["county"]
if kwargs["city_name"] not in item["addr_full"]:
    item["addr_full"] = item["addr_full"] + kwargs["city_name"]
```